src/Web/Sfa.Das.Sas.Infrastructure/Mapping/ProviderNameSearchMapping.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using Sfa.Das.Sas.Core.Domain.Model;
+
+namespace Sfa.Das.Sas.Infrastructure.Mapping
+{
+    public class ProviderNameSearchMapping : IProviderNameSearchMapping
+        {
+
+        public IEnumerable<ProviderNameSearchResult> FilterNonMatchingAliases(string searchTerm, IEnumerable<ProviderNameSearchResult> resultsToFilter)
```

Comment:
While the tests all still pass with this change, when running it keeps erroring. And because you can't put a yield return in a try catch, I cannot get to the bottom of what the issue is.  I assumed it was the item recasting back to itself, but even when I created a new variable using the item's details to hydrate it, it still kept erroring.  

I think it's readable as it is, so I'll do some work to integrate the linq statement in (as it's neater), but otherwise keep the code as it.